relayer/src/message_relayer/common/gear/block_listener.rs

```diff
@@ -46,12 +47,10 @@ impl BlockListener {
     pub fn run<const RECEIVER_COUNT: usize>(self) -> [Receiver<GearBlockNumber>; RECEIVER_COUNT] {
         let (senders, receivers): (Vec<_>, Vec<_>) = (0..RECEIVER_COUNT).map(|_| channel()).unzip();
 
-        tokio::spawn(async move {
-            loop {
-                let res = self.run_inner(&senders).await;
-                if let Err(err) = res {
-                    log::error!("Gear block listener failed: {}", err);
-                }
+        tokio::task::spawn_blocking(move || loop {
```

Comment:
Why `spawn_blocking`?

Reply:
Like it is in other places: CheckpointsExtractor and MessageSender